helpers.ts

```diff
@@ -388,6 +388,49 @@ export async function connectEventuallyUntilTimeout(factory: () => net.Socket, t
 	});
 }
 
+export function isPortAvailable(candidatePort: number): Promise<boolean> {
+	return new Promise<boolean>((resolve, reject) => {
+		let isResolved = false;
+		let server = net.createServer();
+
+		server.on("error", (err: Error) => {
+			if (!isResolved) {
+				isResolved = true;
+				resolve(false);
+			}
+		});
+
+		server.once("close", () => {
+			if (!isResolved) { // "close" will be emitted right after "error"
+				isResolved = true;
+				resolve(true);
+			}
+		});
+
+		server.on("listening", (err: Error) => {
+			if (err && !isResolved) {
+				isResolved = true;
+				resolve(false);
+			}
+
+			server.close();
+		});
+
+		server.listen(candidatePort, "localhost");
+	});
+}
+
+export async function getAvailablePort(port: number): Promise<number> {
```

Comment:
We want to have a method which can get available port in some range. In the debug service in {N} CLI we want to check for free port starting from a specific port because we want to start chrome devtools on one port every time if it is possible. This way the users will be able to just refresh the chrome devtools page when using the debug command again.